srunner/scenarios/actor_flow.py

```diff
@@ -319,3 +328,93 @@ def __del__(self):
+    It will remove the BackgroundActivity from the lane where ActorFlow starts.
+    """
+
+    def __init__(self, world, ego_vehicles, config, randomize=False, debug_mode=False, criteria_enable=True,
+                 timeout=180):
+        """
+        Setup all relevant parameters and create scenario
+        and instantiate scenario manager
+        """
+        self._world = world
+        self._map = CarlaDataProvider.get_map()
+        self.timeout = timeout
+
+        self._start_actor_flow = convert_dict_to_location(config.other_parameters['start_actor_flow'])
+        self._end_actor_flow = convert_dict_to_location(config.other_parameters['end_actor_flow'])
+        self._sink_distance = 2
+
+        self._end_distance = 40
+
+        if 'flow_speed' in config.other_parameters:
+            self._flow_speed = float(config.other_parameters['flow_speed']['value'])
+        else:
+            self._flow_speed = 10 # m/s
+
+        if 'source_dist_interval' in config.other_parameters:
+            self._source_dist_interval = [
+                float(config.other_parameters['source_dist_interval']['from']),
+                float(config.other_parameters['source_dist_interval']['to'])
+            ]
+        else:
+            self._source_dist_interval = [5, 7] # m
+
+
+        super(CleanCrossActorFlow, self).__init__("CleanCrossActorFlow",
+                                             ego_vehicles,
+                                             config,
+                                             world,
+                                             debug_mode,
+                                             criteria_enable=criteria_enable)
+
+    def _create_behavior(self):
+        """
+        Vehicles run from the start to the end continuously.
+        """
+        source_wp = self._map.get_waypoint(self._start_actor_flow)
+        sink_wp = self._map.get_waypoint(self._end_actor_flow)
+
+        grp = GlobalRoutePlanner(CarlaDataProvider.get_map(), 2.0)
+        route = grp.trace_route(source_wp.transform.location, sink_wp.transform.location)
+        junction_id = None
+        for wp, _ in route:
+            if wp.is_junction:
+                junction_id = wp.get_junction().id
+                break
+
+        root = py_trees.composites.Parallel(
+            policy=py_trees.common.ParallelPolicy.SUCCESS_ON_ONE)
+        root.add_child(ActorFlow(
+            source_wp, sink_wp, self._source_dist_interval, self._sink_distance, self._flow_speed))
+        end_condition = py_trees.composites.Sequence()
+        end_condition.add_child(WaitEndIntersection(self.ego_vehicles[0], junction_id))
+        
+        root.add_child(end_condition)
+
+        sequence = py_trees.composites.Sequence()
+
+        if self.route_mode:
+            sequence.add_child(RemoveLane(str(source_wp.lane_id)))
+            sequence.add_child(JunctionScenarioManager('opposite', True))
+            sequence.add_child(StopEntries())
```

Comment:
Up to discussion, but the StopEntries might not be needed for this type of scenario, as it stops all non route entries, which the highway doesn't have, as the intersection is just the highway exit.

Reply:
I didn't notice this. I think both JunctionScenarioManager and StopEntries are unnecessary since this scenario has no junctions.